src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java

```diff
@@ -33,34 +47,52 @@ public int getPriority() {
     }
 
     @Override
-    public void init(FilterConfig filterConfig) throws ServletException {}
+    public void init(FilterConfig filterConfig) throws ServletException {
+        GeoServerDataDirectory dataDirectory =
+                GeoServerExtensions.bean(GeoServerDataDirectory.class);
+        if (dataDirectory != null) {
+            USE_AS_SERVLET_FILTER = true;
+            Resource resource = dataDirectory.get(RulesDao.getRulesPath());
+            rules = RulesDao.getRules(resource.in());
+            resource.addListener(notify -> rules = RulesDao.getRules(resource.in()));
+        }
+    }
 
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
             throws IOException, ServletException {
-        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
-        if (httpServletRequest.getRequestURI().contains("web/wicket")
-                || httpServletRequest.getRequestURI().contains("geoserver/web")) {
-            chain.doFilter(request, response);
-            return;
-        }
-        UrlTransform urlTransform =
-                new UrlTransform(
-                        httpServletRequest.getRequestURI(), httpServletRequest.getParameterMap());
-        String originalRequest = urlTransform.toString();
-        rules.forEach(rule -> rule.apply(urlTransform));
-        if (urlTransform.haveChanged()) {
-            Utils.info(
-                    LOGGER,
-                    "Request '%s' transformed to '%s'.",
-                    originalRequest,
-                    urlTransform.toString());
-            chain.doFilter(new RequestWrapper(urlTransform, httpServletRequest), response);
+        if (isEnabled()) {
+            HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+            if (httpServletRequest.getRequestURI().contains("web/wicket")
+                    || httpServletRequest.getRequestURI().contains("geoserver/web")) {
+                chain.doFilter(request, response);
+                return;
+            }
+            UrlTransform urlTransform =
+                    new UrlTransform(
+                            httpServletRequest.getRequestURI(),
+                            httpServletRequest.getParameterMap());
+            String originalRequest = urlTransform.toString();
+            rules.forEach(rule -> rule.apply(urlTransform));
+            if (urlTransform.haveChanged()) {
+                Utils.info(
+                        LOGGER,
+                        "Request '%s' transformed to '%s'.",
+                        originalRequest,
+                        urlTransform.toString());
+                chain.doFilter(new RequestWrapper(urlTransform, httpServletRequest), response);
+            } else {
+                chain.doFilter(request, response);
```

Comment:
Code is getting a bit convoluted with 3 places that call chain.doFilter... could it be flipped so that there is a single one calling onto doFilter, and the one that is active, which just replaces request with the wrapper?